compiler-rt/lib/mpk_untrusted/mpk.cpp

```diff
@@ -0,0 +1,113 @@
+// mpk.cpp
+//
+// Copyright 2018 Paul Kirth
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+
+// This file supplies a set of APIs similar or identical to those found in
+// glibc-2.27 We mimic these for future compatibility with standard libraries.
+
+// set to 1 if we can use the same implementation for pkey_mprotect as
+// glibc-2.27
```

Comment:
so this is definitely my code, but mixing licenses here is weird.  I don't know how much we actually need these mpk things now that GLIBC has proper implementations. It can stay for now, unless you think its redundant.